src/cache.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ not
+ * use this file except in compliance with the License. You may obtain a copy
+ of
+ * the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ under
+ * the License.
+ */
+
+'use strict';
+
+import * as Datastore from '@google-cloud/datastore';
+import {DatastoreKey} from '@google-cloud/datastore/entity';
+import * as Koa from 'koa';
+
+const datastore: Datastore = new (Datastore as any)();
```

Comment:
what if you made this in the DataStore cache constructor instead of making a singleton?